quickwit/quickwit-control-plane/src/ingest/ingest_controller.rs

```diff
@@ -254,132 +319,96 @@ impl IngestController {
+                num_missing_shards_per_source_uids
+                    .entry(source_uid)
+                    .and_modify(|num_missing_shards| *num_missing_shards += 1)
+                    .or_insert(1);
+            }
+        }
+
+        if let Err(metastore_error) = self
+            .try_open_shards(
+                num_missing_shards_per_source_uids,
+                model,
                 &unavailable_leaders,
-            ) else {
-                let get_or_create_open_shards_failure = GetOrCreateOpenShardsFailure {
-                    subrequest_id: get_open_shards_subrequest.subrequest_id,
-                    index_id: get_open_shards_subrequest.index_id,
-                    source_id: get_open_shards_subrequest.source_id,
-                    reason: GetOrCreateOpenShardsFailureReason::SourceNotFound as i32,
-                };
-                get_or_create_open_shards_failures.push(get_or_create_open_shards_failure);
-                continue;
-            };
-            if !open_shard_entries.is_empty() {
-                let open_shards: Vec<Shard> = open_shard_entries
-                    .into_iter()
-                    .map(|shard_entry| shard_entry.shard)
-                    .collect();
-                let get_or_create_open_shards_success = GetOrCreateOpenShardsSuccess {
-                    subrequest_id: get_open_shards_subrequest.subrequest_id,
-                    index_uid: index_uid.into(),
-                    source_id: get_open_shards_subrequest.source_id,
-                    open_shards,
-                };
-                get_or_create_open_shards_successes.push(get_or_create_open_shards_success);
+                progress,
+            )
+            .await
+        {
+            // We experienced a metastore error. If this is not certain abort, we need
+            // to restart the control plane, to make sure the control plane is not out-of-sync.
+            //
+            if !metastore_error.is_transaction_certainly_aborted() {
+                return Err(metastore_error);
             } else {
-                let shard_id = ShardId::from(Ulid::new());
-                let open_shard_subrequest = metastore::OpenShardSubrequest {
-                    subrequest_id: get_open_shards_subrequest.subrequest_id,
-                    index_uid: index_uid.into(),
-                    source_id: get_open_shards_subrequest.source_id,
-                    shard_id: Some(shard_id),
-                    // These attributes will be overwritten in the next stage.
-                    leader_id: "".to_string(),
-                    follower_id: None,
-                };
-                open_shards_subrequests.push(open_shard_subrequest);
+                // If not, let's just log something.
+                // This is not critical. We will just end up return some failure in the response.
+                error!(metastore_error=?metastore_error, "failed to open shards on the metastore");
```

Comment:
```suggestion
                error!(error=?metastore_error, "failed to open shards on the metastore");
```
It's just easier to look for spans with an error attribute.